### ISSUE TITLE
spatial.assign_container signature changed during the same minor version

### ISSUE THREAD

**Issue by roehroeh** [IfcOpenShell/IfcOpenShell#4531]
Hello,

I've encountered a small change in the API that caused quite a bit of confusion at our latest update.
The recent commit 109787d869a218ec5d1f684ed81ead7bf5d8ee71 introduced batching for spatial relationships and changed the keyword accordingly from product to products.
Due to the fact that both versions were released under 0.7.0, my latest build took the new version and complained about 
```Usecase.__init__() got an unexpected keyword argument 'product'```

It's not directly a bug, but I wanted to leave this issue here for everybody who encountered the same error as me and also as a way to inquire about the way the versioning in this project is linked to API stability.

I really enjoy working with the module, so thanks for all the work.

**Comment by Moult** [comment]
Cheers, indeed we are breaking things and don't currently follow semver. Everything is now sorta alpha in the Python world.

There is a plan to adopt semver in a couple versions.

**Comment by aothms** [comment]
We could add a backwards compatibility layer here? https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/api/__init__.py#L33

```py
if 'product' in settings: 
    settings = {k: v for k, v in settings.items() if k != 'product'}} | {'products': [settings['product']]}
```

or update all api implementations to accept both instance | aggregate[instance] while keeping the signatures the same. I would also say that we should adhere to some versioning rules for the API. Considering also there is quite a bit of usage of these.

**Comment by Moult** [comment]
Yeah if its not a lot of work happy to leave it in but would love to remove it as soon as 0.8.0 is out.

**Comment by Moult** [comment]
Ping @Andrej730

**Comment by Andrej730** [comment]
Added support for old arguments. It's also printing warnings like below to get dev attention. 
Not sure when 0.8.0 will be released, so I added a check that will make our tests fail on 01 Jun 24 indicating that those arguments can be deprecated completely. What do you think, that should cover it?

```
WARNING. `product` argument is deprecated for API method "spatial.assign_container" and should be replaced with `products`.
WARNING. `product` argument is deprecated for API method "spatial.unassign_container" and should be replaced with `products`.
WARNING. `product` argument is deprecated for API method "group.unassign_group" and should be replaced with `products`.
WARNING. `item` argument is deprecated for API method "layer.assign_layer" and should be replaced with `items`.
WARNING. `item` argument is deprecated for API method "layer.unassign_layer" and should be replaced with `items`.
```

**Comment by roehroeh** [comment]
I'm blown away by the speed of your reaction. Thanks for including optional arguments, that should work for me and my project
